### DISCUSSION TITLE
Is autoprefixer absolutely necessary?

### DISCUSSION THREAD

**Discussion by martinszeltins** [tailwindlabs/tailwindcss#5543] [Help] [answered]
I am creating an app using Vue 3 + Vite. And I do not understand why do I need autoprefixer? Can I safely remove it or is it like a dependency for Tailwind?

I do not plan to prefix any tailwind classes.

**Comment by adamwathan** [comment, marked as answer]
Hey! For best browser support you do want to use it, because many CSS properties require vendor-specific prefixes to work even in 2021. For example `user-select` doesn't work in Safari without `-webkit-user-select`. Tailwind doesn't handle this for you automatically (unless you use our CLI tool, which has autoprefixer built-in), so for best browser support you will want to include autoprefixer in your PostCSS plugins.

**Reply by martinszeltins** [reply]
> Hey! For best browser support you do want to use it, because many CSS properties require vendor-specific prefixes to work even in 2021. For example `user-select` doesn't work in Safari without `-webkit-user-select`. Tailwind doesn't handle this for you automatically (unless you use our CLI tool, which has autoprefixer built-in), so for best browser support you will want to include autoprefixer in your PostCSS plugins.

Oh, that makes sense. I was surprised that I didn't get any errors from Tailwind when I removed autoprefixer so that made me wonder if it's a strict requirement. Thanks, Adam!

**Reply by timbomckay** [reply]
I did a search in issues for autoprefixer and didn't realize there was a discussion section in GitHub now. I ended up [creating another discussion thread](https://github.com/tailwindlabs/tailwindcss/discussions/5649) before I realized this was here. I provided a little more info there. I'd still be in favor of opting into autoprefixer though.
